book-content/chapters/08-classes.md

```diff
@@ -0,0 +1,1044 @@
+How do we fix this?
+
+### Adding a Constructor
+
+In order to fix these errors, we need to add a `constructor` to the class. The `constructor` is a special method that runs when a new instance of the class is created. It's where you can set up the initial state of the object.
+
+To start, we'll add a constructor that assigns values for the properties of the `Album` class:
+
+```typescript
+class Album {
+  title: string;
+  artist: string;
+  releaseYear: number;
+
+  constructor() {
+    this.title = "Loop Finding Jazz Records";
+    this.artist = "Jan Jelinek";
+    this.releaseYear = 2001;
+  }
+}
+```
+
+Now, when we create a new instance of the `Album` class, we can access the properties and values we've set in the constructor:
+
+```typescript
+const loopFindingJazzRecords = new Album();
+
+console.log(loopFindingJazzRecords.title); // Output: Loop Finding Jazz Records
+```
+
+The `new` keyword creates a new instance of the `Album` class, and the constructor sets the initial values of our class's properties. In this case, because the properties are hardcoded, every instance of the `Album` class will have the same values.
+
+#### You Don't Always Need To Type Class Properties
+
+As we'll see, TypeScript can do some really smart inference with classes. It's able to infer the types of the properties from where we assign them in the constructor, so we can actually drop some of the type annotations:
+
+```typescript
+class Album {
+  title;
+  artist;
+  releaseYear;
+
+  constructor() {
+    this.title = "Loop Finding Jazz Records";
+    this.artist = "Jan Jelinek";
+    this.releaseYear = 2001;
+  }
+}
+```
+
+However, it's common to see the types specified in the class body as well since they act as a form of documentation for the class that's quick to read.
+
+### Adding Arguments to the Constructor
+
+We can use the constructor to declare arguments for the class. This allows us to pass in values when creating a new instance of the class.
+
+Update the constructor to accept an `opts` argument that includes the properties of the `Album` class:
+
+```typescript
+// inside the Album class
+constructor(opts: { title: string; artist: string; releaseYear: number }) {
+ // ...
+}
+```
+
+Then inside of the body of the constructor, we'll use assign `this.title`, `this.artist`, and `this.releaseYear` to the values of the `opts` argument.
+
+```typescript
+// inside the Album class
+constructor(opts: { title: string; artist: string; releaseYear: number }) {
+  this.title = opts.title;
+  this.artist = opts.artist;
+  this.releaseYear = opts.releaseYear;
+}
+```
+
+The `this` keyword refers to the instance of the class, and it's used to access the properties and methods of the class.
+
+Now, when we create a new instance of the `Album` class, we can pass an object with the properties we want to set. If we don't provide any values, the default values will be used:
```

Comment:
you mention default values here and u even rely on them in the latter example but u have no default values here anywhere